roscala/src/main/scala/coop/rchain/rosette/prim/PrimQueue.scala

```diff
@@ -0,0 +1,186 @@
+package coop.rchain.rosette.prim
+
+import coop.rchain.rosette.Ob.ABSENT
+import coop.rchain.rosette.{Ctxt, Fixnum, Ob, Queue, RblBool, Tuple}
+import coop.rchain.rosette.macros.{checkArgumentMismatch, checkTypeMismatch}
+import coop.rchain.rosette.prim.Prim._
+
+object PrimQueue {
+
+  object QueueNew extends Prim {
+    override val name: String = "queue-new"
+    override val minArgs: Int = 0
+    override val maxArgs: Int = 0
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Queue] =
+      Right(Queue.create())
+  }
+
+  object QueueDepth extends Prim {
+    override val name: String = "queue-depth"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 1
+
+    @checkTypeMismatch[Queue]
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Fixnum] = {
+      val q = ctxt.argvec.elem.head.asInstanceOf[Queue]
+      Right(q.nElem.asInstanceOf[Fixnum])
+    }
+  }
+
+  object QueueEmpty extends Prim {
+    override val name: String = "queue-empty?"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 1
+
+    @checkTypeMismatch[Queue]
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, RblBool] = {
+      val q = ctxt.argvec.elem.head.asInstanceOf[Queue]
+      Right(RblBool(q.empty()))
+    }
+  }
+
+
+  object QueueEnQueue extends Prim {
+    override val name: String = "queue-enqueue"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 2
+
+    @checkTypeMismatch[Ob]
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Queue] = {
+      val q = ctxt.argvec.elem.head.asInstanceOf[Queue]
+      val ob = ctxt.argvec.elem(1)
+      q.enQueue(ob)
+      Right(q)
+    }
+  }
+
+  object QueueDeQueue extends Prim {
+    override val name: String = "queue-dequeue"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 1
+
+    @checkTypeMismatch[Ob]
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Ob] = {
+      val q = ctxt.argvec.elem.head.asInstanceOf[Queue]
+      if(q.empty()) {
+        Left(QueueEmptyError("queue is empty"))
```

Comment:
The message inside of the `QueueEmptyError` is not really providing any additional value. Can you change the `QueueEmptyError` to a `case object`?